### PR TITLE
Dockerfile with ROS2 humble and Cartographer

### PR DESCRIPTION
**Pull request by Assume-Zhan**
Closes: #12

### CODE REVIEW

**Review by j3soon — CHANGES_REQUESTED** [review]
As mentioned in our offline discussion, please add an example using multiple lidars.

Thank you!

File: cartographer_ws/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  ```bash
+  ros2 launch multi_lidar_gazebo multi_lidar_gazebo.launch.py
+  ```
+
+#### Run the SLAM node
+
+- Run the cartographer SLAM node in new window of `tmux`
+  ```bash
+  ros2 launch cartographer_demo cartographer_demo.launch.py
+  ```
```
Comment (j3soon) [inline cartographer_ws/README.md:69]: This step cannot visualize the SLAM map in RVIZ, I wonder if there's some bug in this launch file? Or is some settings in the config files incorrectly set?

**Review by j3soon — CHANGES_REQUESTED** [review]
Thanks for the PR!

Learned a lot on setting up cartographer when reviewing.

There are only 2 issues as mentioned below. I think we can merge this after fixing those issues.

File: cartographer_ws/README.md (hunk 1 above)
Comment (Assume-Zhan) [reply cartographer_ws/README.md:69]: I forgot to include the map tunnel in the RViz configuration, which caused the map not to show up. I'll add it in the following commit to rectify this issue.

**Review by j3soon — APPROVED** [review]
Thanks for the PR! Looks good to me~

**Comment by j3soon** [comment]
As an additional note, I think it's better to directly use permalinks in the comment, instead of separating the file path.

See commit https://github.com/j3soon/ros2-essentials/commit/5ba989934b2da077edcfa4aae7b23e57114ff9b2.

The permalink of a file can be copied from: `Hamburger Icon` > `Copy permalink`

Refs:

- https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files
- https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet

**Pull request merged**